R/pmcmc_parameters_nested.R

```diff
@@ -247,73 +237,58 @@ pmcmc_parameters_nested <- R6::R6Class(
         lp_vary <- vnapply(pops,
                            function(x) private$varied_parameters[[x]]$prior(
                              theta[x, private$param_names$varied]))
-      } else {
-        lp_vary <- 0
-      }
+                           } else {
+                             lp_vary <- 0
+                           }
 
       lp_fix + lp_vary
     },
 
-    ##' @description This proposes a new parameter matrix given your current
-    ##' matrix and the variance-covariance matrices of the proposal
-    ##' kernels, discretises any discrete values, and reflects bounded
-    ##' parameters until they lie within `min`:`max`. Returns matrix with rows
-    ##' corresponding to populations and columns to parameters.
-    ##'
-    ##' @param theta a parameter matrix with columns in the same order as
-    ##' `$names()` and rows in the same order as `$populations()`.
-    ##'
-    ##' @param scale an optional scaling factor to apply to the
-    ##' proposal distribution. This may be useful in sampling starting
-    ##' points. The parameter is equivalent to a multiplicative factor
-    ##' applied to the variance covariance matrix.
-    ##'
-    ##' @param type specifies which type of parameters should be proposed,
-    ##' either fixed parameters only ("fixed"), varied only ("varied"), or
-    ##' both ("both") types. For 'fixed' and 'varied', parameters of the
-    ##' other type are left unchanged.
-    propose = function(theta, scale = 1, type = c("fixed", "varied", "both")) {
+##' @description This proposes a new parameter matrix given your current
+##' matrix and the variance-covariance matrices of the proposal
+##' kernels, discretises any discrete values, and reflects bounded
+##' parameters until they lie within `min`:`max`. Returns matrix with rows
+##' corresponding to populations and columns to parameters.
+##'
+##' @param theta a parameter matrix with columns in the same order as
+##' `$names()` and rows in the same order as `$populations()`.
+##'
+##' @param scale an optional scaling factor to apply to the
+##' proposal distribution. This may be useful in sampling starting
+##' points. The parameter is equivalent to a multiplicative factor
+##' applied to the variance covariance matrix.
+##'
+##' @param type specifies which type of parameters should be proposed,
+##' either fixed parameters only ("fixed"), varied only ("varied"), or
+##' both ("both") types. For 'fixed' and 'varied', parameters of the
+##' other type are left unchanged.
+    propose = function(theta, scale = 1, type = c("both", "varied", "fixed")) {
       type <- match.arg(type)
```

Comment:
What do you think about making this required, rather than with a default?

Normally I write a stricter version of match.arg that requires exact matches